src/pages/resultsView/download/DownloadUtils.ts

```diff
@@ -383,6 +457,46 @@ export function generateDownloadFileRows(
     return rows;
 }
 
+export function generateGenericAssayDownloadFileRows(
+    sampleGenericAssayDataByStableId: { [key: string]: GenericAssayData[] },
+    stableIds: string[],
+    sampleIndex: { [sampleKey: string]: Sample },
+    sampleKeys: string[],
+    extractValue?: (alteration: GenericAssayData) => string
+): { [sampleKey: string]: IDownloadFileRow } {
+    const rows: { [sampleKey: string]: IDownloadFileRow } = {};
+
+    sampleKeys.forEach(sampleKey => {
```

Comment:
will there sometimes be members of sampleIndex which are NOT in sampleKeys?

Reply:
maybe sampleKeys should be first argument since it specifies the composition of return?

Reply:
Sure!